app/src/main/java/org/xbmc/kore/ui/AbstractSearchableFragment.java

```diff
@@ -0,0 +1,207 @@
+     * For instance in {@link #onAttach(Activity)}
+     *
+     * @param supportsSearch true if you support search queries, false otherwise
+     */
+    public void setSupportsSearch(boolean supportsSearch) {
+        this.supportsSearch = supportsSearch;
+    }
+
+    /**
+     * Save the search state of the list fragment
+     */
+    public void saveSearchState() {
+        savedSearchFilter = searchFilter;
+    }
+
+    /**
+     * @return text entered in searchview
+     */
+    public String getSearchFilter() {
+        return searchFilter;
+    }
+
+    private void setupSearchMenuItem(Menu menu, MenuInflater inflater) {
+        inflater.inflate(R.menu.media_search, menu);
+        MenuItem searchMenuItem = menu.findItem(R.id.action_search);
+        if (searchMenuItem != null) {
+            searchView = (SearchView) MenuItemCompat.getActionView(searchMenuItem);
+            searchView.setOnQueryTextListener(this);
+            searchView.setQueryHint(getString(R.string.action_search));
+            if (!TextUtils.isEmpty(savedSearchFilter)) {
+                searchMenuItem.expandActionView();
+                searchView.setQuery(savedSearchFilter, false);
+                //noinspection RestrictedApi
+                searchView.clearFocus();
+            }
+
+            MenuItemCompat.setOnActionExpandListener(searchMenuItem, new MenuItemCompat.OnActionExpandListener() {
+                @Override
+                public boolean onMenuItemActionExpand(MenuItem item) {
+                    return true;
+                }
+
+                @Override
+                public boolean onMenuItemActionCollapse(MenuItem item) {
+                    searchFilter = savedSearchFilter = null;
+                    refreshList();
+                    return true;
+                }
+            });
+        }
+
+        //Handle clearing search query using the close button (X button).
+        View view = searchView.findViewById(R.id.search_close_btn);
+        if (view != null) {
+            view.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    EditText editText = (EditText) searchView.findViewById(R.id.search_src_text);
+                    editText.setText("");
+                    searchView.setQuery("", false);
+                    searchFilter = savedSearchFilter = "";
+                    refreshList();
+                }
+            });
+        }
+    }
+
+
+    /**
+     * Search view callbacks
+     */
+    /** {@inheritDoc} */
+    @Override
+    public boolean onQueryTextChange(String newText) {
+        if ((!searchView.hasFocus()) && TextUtils.isEmpty(newText)) {
+            //onQueryTextChange called as a result of manually expanding the searchView in setupSearchMenuItem(...)
+            return true;
+        }
+
+        /**
+         * When this fragment is paused, onQueryTextChange is called with an empty string.
```

Comment:
Please update this comment as it is referring to a `cursoradapter` that is not used here.
If I understand correctly I think the first two sentences will do just fine.